src/core/mac/mac_links.hpp

```diff
@@ -457,21 +457,27 @@ class Links : public InstanceLocator
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
     /**
-     * This method transitions all radios link to CSL sample state.
+     * This method notifies all radios about CSL parameters change in order to start/stop CSL sampling.
```

Comment:
If I understand the change in thsi PR, your goal is to add common `UpdateCslState()` which notifies of changes to any of the CSL parameter and then signal radio `Get<Radio>().EnableCsl()` and update all variables.

I think we should keep this separate from `SubMac::CslSample()` (which asks sub-mac to enter CSL sample state).

My suggestion is to see if we can add this logic in `SubMac` as a method `UpdateCslState()`...which can be exposed upwards through `Links` and `Mac` as method that call the `SubMac` one.
- It would update the member variable (CSL period/channel) 
- Calls `Get<Radio>().EnableCsl()` if needed
- Update its local state if it is already in CSL sample state.

`SubMac` is closer to `Radio` and calls other `Radio` methods, so it seems to fit well for it to also be in charge of calling `EnableCsl()`.

Reply:
This proposal is not so trivial, for instance `Get<Radio>().EnableCsl()` requires the parent information, but SubMac doesn't keep this.

Reply:
Can we pass down the parent address in `SubMac::UpdateCsl()` call as a parameter?